### PR TITLE
Move to new eldoc

### PR DESCRIPTION
**Pull request by yyoncho**
- continue #3082

### CODE REVIEW

**Review by ericdallo — APPROVED** [review]
I don't see any problems with that.

We need to remove the check for these removed version on github settings after merge

File: lsp-mode.el [hunk 1]
```diff
@@ -4851,10 +4837,33 @@ If INCLUDE-DECLARATION is non-nil, request the server to include declarations."
...
    (lsp-notify "$/cancelRequest" `(:id ,id))))
 
-(defun lsp-eldoc-function ()
-  "`lsp-mode' eldoc function."
-  (run-hooks 'lsp-eldoc-hook)
-  eldoc-last-message)
+(defvar-local lsp--hover-saved-bounds nil)
+
+(defun lsp-eldoc-function (cb &rest _ignored)
+  "`lsp-mode' eldoc function to display hover info (based on `textDocument/signatureHelp')."
```
Comment (michaelpj) [inline lsp-mode.el:4843]:
```suggestion
  "`lsp-mode' eldoc function to display hover info (based on `textDocument/hover')."
```

**Review by jcs090218 — APPROVED** [review]
IMO, we can drop the entire Emacs 26.x. Esp. without the native json support; it's quite slow on windows machines.

File: lsp-mode.el (hunk 1 above)
Comment (yyoncho) [reply lsp-mode.el]: Applied, thanks.

**Comment by yyoncho** [comment]
> IMO, we can drop the entire Emacs 26.x. Esp. without the native json support; it's quite slow on windows machines.

I don't think that we do anything special to support emacs 26 (disregarding the native json stuff, but you can still compile emacs 27 without native json). Let's drop emacs 26.3 when it turns into a burden.

**Pull request merged**